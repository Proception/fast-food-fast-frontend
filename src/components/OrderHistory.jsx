import React, { Component } from 'react';
import { connect } from 'react-redux';
import OrderHistoryCards from './OrderHistoryCards.jsx';
import { getUserOrders } from '../actions/orderActions';

export class OrderHistory extends Component {

  componentDidMount(){
    this.props.getUserOrders();
  }
  generateOrders(){

    const { userOrders } = this.props.order;
    userOrders.sort(function(a,b){
      return new Date(b.order_date) - new Date(a.order_date);
    });
    if (!Array.isArray(userOrders) || !userOrders.length) {
      return ( 
        <div class="container">
            <div class="old-order">
              User has not made any Order Yet
            </div>
          </div> )
    }else {
      return <OrderHistoryCards userOrders={userOrders}/>
    }
    
  }
  render() {
    return (
      <div>
        <section class="order">
          {
            this.generateOrders()
          }
        </section>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  auth: state.auth,
  order: state.order,
});

export default connect(mapStateToProps, { getUserOrders })(OrderHistory);