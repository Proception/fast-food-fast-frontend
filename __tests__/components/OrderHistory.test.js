import React from 'react';
import { shallow } from 'enzyme';
import OrderHistory from '../../src/components/OrderHistory.jsx';

let wrapper;
beforeEach(() => {
  wrapper = shallow(<OrderHistory />);
});
describe('OrderHistory Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
