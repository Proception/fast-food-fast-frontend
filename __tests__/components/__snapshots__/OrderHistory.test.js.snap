// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderHistory Component should match snapshot 1`] = `
<div>
  <section
    class="order"
  >
    <OrderHistoryCards
      orders={
        Array [
          Object {
            "id": 1,
            "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
            "orderDate": "30 August, 2016",
            "orderNo": "12",
            "orderStatus": "Pending",
            "price": 5000,
          },
          Object {
            "id": 2,
            "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
            "orderDate": "Boiled Yam",
            "orderNo": "54",
            "orderStatus": "In Progress",
            "price": 1200,
          },
          Object {
            "id": 3,
            "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
            "orderDate": "30 October, 2017",
            "orderNo": "67",
            "orderStatus": "Fulfilled",
            "price": 1500,
          },
          Object {
            "id": 4,
            "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
            "orderDate": "30 January, 2018",
            "orderNo": "33",
            "orderStatus": "Completed",
            "price": 3000,
          },
        ]
      }
    />
    <div
      class="pagination"
    >
      <a
        href="#"
      >
        «
      </a>
      <a
        class="active"
        href="#"
      >
        1
      </a>
      <a
        href="#"
      >
        2
      </a>
      <a
        href="#"
      >
        »
      </a>
    </div>
  </section>
</div>
`;
